### DISCUSSION TITLE
Context arguments

### DISCUSSION THREAD

**Discussion by JamesHemery** [joe-bell/cva#162] [Ideas]
Hi there, 

As a follow-up to #161, we have also implemented a context feature in our internal fork.

Unlike the other discussion (#161), I think this feature is much more debatable and optional.

What I suggest : 

```js
const button = cva('rounded p-4', {
  variants: {
    color: {
      primary: ({ disabled }) => disabled ? 'bg-green-400' : 'bg-green-900',
      secondary: ({ disabled }) => disabled ? 'bg-blue-400' : 'bg-blue-900',
    }
  },
  defaultVariants: {
    color: 'primary',
  },
});

// second argument is context
const style = button({ color: 'primary' }, { disabled: true }); // rounded p-4 bg-green-900
```

I am not sure that this is the desired direction for the bookshop though, so I am opening a side discussion.

Thanks,

**Comment by joe-bell** [comment]
Are you suggesting this as a replacement of compoundVariants?

I’m not too sure about the idea of introducing two approaches to the same challenge

**Reply by JamesHemery** [reply]
Not really, in our case we consider that some elements of context (like disabled or loading boolean) are not variants, that's why we thought to introduce a context system.

The idea may be bad, that's why we hesitate.

**Reply by joe-bell** [reply]
In that case I don’t think I’ll go forward with this idea for now, but appreciate the suggestion
